airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -384,39 +384,38 @@ public WebBackendConnectionRead webBackendUpdateConnectionNew(final WebBackendCo
       throws ConfigNotFoundException, IOException, JsonValidationException {
     final List<UUID> operationIds = updateOperations(webBackendConnectionUpdate);
     final ConnectionUpdate connectionUpdate = toConnectionUpdate(webBackendConnectionUpdate, operationIds);
-
     final UUID connectionId = webBackendConnectionUpdate.getConnectionId();
-
     final ConfiguredAirbyteCatalog existingConfiguredCatalog =
         configRepository.getConfiguredCatalogForConnection(connectionId);
-    final io.airbyte.protocol.models.AirbyteCatalog existingCatalog = CatalogHelpers.configuredCatalogToCatalog(existingConfiguredCatalog);
-    final AirbyteCatalog apiExistingCatalog = CatalogConverter.toApi(existingCatalog);
-    final AirbyteCatalog newAirbyteCatalog = webBackendConnectionUpdate.getSyncCatalog();
-    final CatalogDiff catalogDiff = connectionsHandler.getDiff(apiExistingCatalog, newAirbyteCatalog);
-
-    final List<StreamDescriptor> apiStreamsToReset = getStreamsToReset(catalogDiff);
-    List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset =
-        apiStreamsToReset.stream().map(ProtocolConverters::streamDescriptorToProtocol).toList();
-
     ConnectionRead connectionRead;
     connectionRead = connectionsHandler.updateConnection(connectionUpdate);
 
-    if (!streamsToReset.isEmpty()) {
-      final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody().connectionId(connectionId);
-      final ConnectionStateType stateType = getStateType(connectionIdRequestBody);
+    if(!webBackendConnectionUpdate.getSkipReset()) {
+      final io.airbyte.protocol.models.AirbyteCatalog existingCatalog = CatalogHelpers.configuredCatalogToCatalog(existingConfiguredCatalog);
+      final AirbyteCatalog apiExistingCatalog = CatalogConverter.toApi(existingCatalog);
+      final AirbyteCatalog newAirbyteCatalog = webBackendConnectionUpdate.getSyncCatalog();
+      final CatalogDiff catalogDiff = connectionsHandler.getDiff(apiExistingCatalog, newAirbyteCatalog);
 
-      if (stateType == ConnectionStateType.LEGACY || stateType == ConnectionStateType.NOT_SET || stateType == ConnectionStateType.GLOBAL) {
-        streamsToReset = configRepository.getAllStreamsForConnection(connectionId);
+      final List<StreamDescriptor> apiStreamsToReset = getStreamsToReset(catalogDiff);
+      List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset =
```

Comment:
We also need Benoit's configuration diff change from here for this to be fully working as desired: https://github.com/airbytehq/airbyte/pull/14626/files

Reply:
Is it ok if these go in separately?

Reply:
Yeah I think it's okay, I just wanted to call out that if someone tries to use this endpoint before Benoits change is in then changes to only the configuration of streams will not result in a reset